src/pages/SettingsCurrency/components/CurrencyPicker/CurrencyPicker.js

```diff
@@ -0,0 +1,79 @@
+// @flow strict
+
+import React from 'react'
+import { t } from 'ttag'
+import { map } from 'lodash-es'
+
+import {
+  JIcon,
+} from 'components/base'
+
+import {
+  JPickerBody,
+  JPickerList,
+  JPickerCurrent,
+  DefaultItem,
+} from 'components/base/JPicker'
+
+import currenciesList from 'data/currencies'
+
+type Props = {|
+  +meta: FinalFormMeta,
+  +input: FinalFormInput,
+|}
+
+export function CurrencyPicker({
+  meta,
+  input,
+}: Props) {
+  const {
+    value,
+    onFocus: handleFocus,
+    onBlur: handleBlur,
+    onChange: handleChange,
+  } = input
+
+  const {
+    code: activeCode = '',
+    name: activeName = '',
+  } = currenciesList[value] || {}
+
+  return (
+    <JPickerBody
```

Comment:
Simple pickers become too complex :( Let's book some time and discuss how to simplify them?